localstack/services/apigateway/apigateway_listener.py

```diff
@@ -804,11 +804,17 @@ def invoke_rest_api_integration_backend(
                 template = integration["requestTemplates"][APPLICATION_JSON]
                 account_id, queue = uri.split("/")[-2:]
                 region_name = uri.split(":")[3]
-
-                new_request = "%s&QueueName=%s" % (
-                    aws_stack.render_velocity_template(template, data),
-                    queue,
-                )
+                if "GetQueueUrl" in template or "CreateQueue" in template:
+                    new_request = "%s&QueueName=%s" % (
+                        aws_stack.render_velocity_template(template, data),
+                        queue,
+                    )
+                else:
+                    queue_url = "{}/{}/{}".format(config.get_edge_url(), account_id, queue)
+                    new_request = "%s&QueueUrl=%s" % (
+                        aws_stack.render_velocity_template(template, data),
+                        queue_url,
+                    )
```

Comment:
Yes, but whenever the string gets too long for the line I usually use .format because the line splitting looks neater imo. But I can of course ditch that. The rest is old code I simply didn't change, should I always overhaul a section like this when I am working on it?

Reply:
:heavy_check_mark: